userspace/libsinsp/plugin.h

```diff
@@ -166,6 +177,10 @@ class sinsp_plugin
 	std::string get_init_schema(ss_plugin_schema_type& schema_type) const;
 	bool set_config(const std::string& config);
 	std::vector<metrics_v2> get_metrics() const;
+	bool capture_open();
+	bool capture_close();
+	thread_pool::routine_id_t subscribe_routine(ss_plugin_routine_fn_t routine_fn, ss_plugin_routine_state_t* routine_state);
+	bool unsubscribe_routine(thread_pool::routine_id_t routine_id);
```

Comment:
Again, I'd put these as private methods

Reply:
Please take a look at `plugin_subscribe_routine` and `plugin_unsubscribe_routine` in `plugin.cpp` (https://github.com/falcosecurity/libs/pull/1674/commits/589affa6e47da669f51e4bbc76cc99f44a2cb701).

Those are the functions set in `ss_plugin_routine_vtable`. 
In those function we cast `ss_plugin_owner_t*` to get the plugin wrapper and access its methods. 
That's why we need `plugin_subscribe_routine` and `plugin_unsubscribe_routine` to be public.